app/user.go

```diff
@@ -2264,15 +2266,61 @@ func (a *App) ConvertBotToUser(bot *model.Bot, userPatch *model.UserPatch, sysad
 }
 
 func (a *App) GetThreadsForUser(userID, teamID string, options model.GetUserThreadsOpts) (*model.Threads, *model.AppError) {
-	threads, err := a.Srv().Store.Thread().GetThreadsForUser(userID, teamID, options)
-	if err != nil {
+	var result model.Threads
+	var eg errgroup.Group
+
+	eg.Go(func() error {
+		totalUnreadThreads, err := a.Srv().Store.Thread().GetTotalUnreadThreads(userID, teamID, options)
+		if err != nil {
+			return errors.Wrapf(err, "failed to count unread threads for user id=%s", userID)
+		}
+		result.TotalUnreadThreads = totalUnreadThreads
```

Comment:
@agnivade, can you sanity check that it's safe to write to /different/ struct properties from multiple goroutines. Intuitively, I think this is safe, but want to make sure!

Reply:
Yes, it's safe. To confirm, you can run mattermost in `-race` mode and see if there are any issues.